File: swift/ValidMountainArray.swift
// https://leetcode.com/problems/valid-mountain-array/

class Solution {
    func validMountainArray(_ arr: [Int]) -> Bool {
        var i = 0
        
        while i < arr.count - 1 && arr[i] < arr[i + 1] {
            i += 1
        }
        
        if i == 0 || i == arr.count - 1 {
            return false
        }
        
        while i < arr.count - 1 && arr[i] > arr[i + 1] {
            i += 1
        }
        
        return i == arr.count - 1
    }
}

File: c/ValidMountainArray.c
// https://leetcode.com/problems/valid-mountain-array/

bool validMountainArray(int* arr, int arrSize){
    if (arrSize <= 2)
        return false;
    int inc = 0, dec = 0;
    for (int i = 1; i < arrSize - 1; i++) {
        if (arr[i-1] < arr[i] && arr[i] > arr[i+1])
            inc++;
        else if (arr[i-1] > arr[i] && arr[i] < arr[i+1])
            dec++;
        else if (arr[i-1] == arr[i] || arr[i] == arr[i+1])
            return false;
    }
    return inc == 1 && dec == 0;
}